### PR TITLE
DB-11145 fix ExternalTableIT Spark3.0 failing tests

### PR DESCRIPTION
**Pull request by martinrupp**

### CODE REVIEW

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by martinrupp** [comment]
jenkins please test branch @dbaas3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OperatorToString.java [hunk 1]
```diff
@@ -328,7 +328,10 @@ else if (operand instanceof ExtractOperatorNode) {
                     if (functionName.equals("SECOND") || functionName.equals("WEEK")) {
                         throwNotImplementedError();
                     } else if (functionName.equals("WEEKDAY")) {
-                        return format("cast(date_format(%s, \"u\") as int) ", opToString2(uop.getOperand(), vars));
+                        if( getSparkVersion().getMajorVersionNumber() >= 3 )
```
Comment (hatyo) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OperatorToString.java:331]: I would be more defensive here and change the line to something like:

> if( getSparkVersion() != null && getSparkVersion().getMajorVersionNumber() >= 3 )

**Review by hatyo — APPROVED** [review]

**Review by ipraznik-splice — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**